### PR TITLE
Copy GDAL XML layer datasources

### LINKED ISSUE #2: oq-consolidate should copy gdal xml
Like the one used for basemaps we have in the Colombia project, otherwise we miss backgrounds in print composers

### PR DESCRIPTION
**Pull request by daniviga**
This PR add support for XML GDAL datasources. We are using this to get TMS layers in QGIS and the print composer.

Fixes #2

### CODE REVIEW

File: consolidatethread.py [hunk 1]
```diff
@@ -82,12 +83,17 @@ def run(self):
...
             layerType = layer.type()
+            layerName = layer.name()
+            layerUri = layer.dataProvider().dataSourceUri()
             if layerType == QgsMapLayer.VectorLayer:
-                layerName = layer.name()
                 # Always convert to GeoPackage
-                self.copyGenericVectorLayer(e, layer, layerName)
+                self.convertGenericVectorLayer(e, layer, layerName)
+            elif (layerType == QgsMapLayer.RasterLayer
+                  and layerUri.endswith('.xml')):
```
Comment (daniviga) [inline consolidatethread.py]: Not sure if there's a better way to identify this kind of layers

File: consolidatethread.py [hunk 2]
```diff
@@ -82,12 +83,17 @@ def run(self):
 
         for layer in layers:
             layerType = layer.type()
```
Comment (ptormene) [inline consolidatethread.py]: We might add here:
```python
if not layer.isValid():
    print("Layer %s is invalid" % layer.name())
elif layerType == ...
```

Comment (daniviga) [reply consolidatethread.py]: Done

File: qconsolidatedialog.py [hunk 3]
```diff
@@ -63,6 +84,16 @@ def setOutDirectory(self):
         self.leOutputDir.setText(outDir)
 
     def accept(self):
+        project_name = self.project_name_le.text()
+        if project_name.endswith('.qgs'):
+            project_name = project_name[:-4]
+        if not project_name:
+            QMessageBox.warning(self,
+                                self.tr("QConsolidate: Error"),
```
Comment (ptormene) [inline qconsolidatedialog.py]: In this copy/paste (my fault) I forgot to rename QConsolidate -> OQ-Consolidate

**Review by ptormene — APPROVED** [review]
Apart from a minor comment, LGTM

**Pull request merged**